web/backend-fastapi/app/controllers/analytics.py

```diff
@@ -0,0 +1,31 @@
+from fastapi import APIRouter, Request, HTTPException
+import os
+import json
+
+router = APIRouter()
+
+@router.post("/saveAnalytics")
+async def save_analytics(request: Request):
+    try:
+        analytics_data = await request.json()
+        
+        # Extract sessionId from the analytics_data
+        session_id = analytics_data.get('sessionId')
+        
+        if not session_id:
+            raise HTTPException(status_code=400, detail="sessionId is required in the analytics data")
+
+        # Create a filename based on the sessionId
+        file_name = f"analytics_{session_id}.json"
```

Comment:
shouldn't we have one log file where it keeps writing to instead of having individual files for session?

Reply:
my concern is the file will be eventually become too large and it could waste resources loading the same data over and over again. with multiple files, only new session files will be loaded

Reply:
The file should be loaded/opened only once during the fastapi start and then it keeps writing to it. Every day or during the bedrock key exchange (once in two days) time we can have a scheduler that backups the logs so the file does not grow too large over time.

Reply:
for now it is fine. we can improve on it later.

Reply:
updated now to put all in one file